src/Makefile

```diff
@@ -1,123 +1,113 @@
-# makefile for pcap2flow 
+# Makefile for joy binaries 
+#
+# Builds binary files for joy open source modules. 
 #
 # Copyright (c) 2016 Cisco Systems 
 
+##
 # variables 
-#
+##
 
+##
 # CWD is the name of the current directory
-#
+##
 TMP = $(shell pwd)
 CWD = $(shell basename $(TMP))
 
-# The version variable identifies the version number of the source
-# code (and not the version of the protocol).
-#
-version = $(shell cat VERSION)
-
-# The sysname variable identifies the system by kernel name, and is
-# used to define a C preprocessor symbol (in the CFLAGS variable) that
-# indicates the operating system type.  Note that it does *not* affect
-# the endianness choice, which is picked up from <sys/types.h>.
-#
-sysname = $(shell uname -s | tr "[:lower:]" "[:upper:]" )
-
+##
 # required libraries for pcap2flow
-#
+##
 LIBS = -lpcap    # packet capture library; hard requirement
 LIBS += -lm      # math library; logf() used in entropy computation
 LIBS += -lcrypto # openSSL crypto library; used in anonymization
 LIBS += -lz
 
-INCLUDEDIR = 
+##
+# check for additional paths to use searching for header files
+##
+ifndef SSLPATH
+INCLUDEDIR = -I ./include
+else
+INCLUDEDIR = -I ./include $(SSLPATH)
+endif
 
-CDEFS     = -D$(sysname)=1 -DVERSION=\"$(version)\"
+##
+# check for additional paths to use searching for library files
+##
+ifndef LIBPATH
+LIBRARYPATH = 
+else
+LIBRARYPATH = $(LIBPATH)
+endif
 
+##
+# compiler directives
+##
+CDEFS     = -D$(SYS_NAME)=1 -DVERSION=\"$(JOY_VERSION)\"
 CFLAGS = -Wall -Wno-deprecated-declarations -g -O3  # -pg # -g # -fstack-protector-all 
 
-NFV9_FILES = exporter.c nfv9.c nfv9.h Makefile
-TLS_FILES = tls.c tls.h
-CLASSIFY_FILES = classify.c classify.h
-
+##
+# variables to make source file handling easier
+##
 PCAP2FLOW_SRC = p2f.c config.c osdetect.c anon.c pkt_proc.c nfv9.c tls.c classify.c radix_trie.c hdr_dsc.c procwatch.c addr_attr.c addr.c wht.c http.c str_match.c acsm.c dns.c example.c
-PCAP2FLOW_HDR = osdetect.h anon.h p2f.h pkt.h tls.h pkt_proc.h radix_trie.h classify.h hdr_dsc.h addr_attr.h addr.h err.h dns.h
+JFDANON_SRC = anon.c addr.c str_match.c acsm.c
+ALL_HEADER_FILES = acsm.h config.h hdr_dsc.h osdetect.h procwatch.h addr.h dns.h http.h output.h radix_trie.h addr_attr.h err.h map.h p2f.h str_match.h anon.h example.h modules.h pkt.h tls.h classify.h feature.h nfv9.h pkt_proc.h wht.h
```

Comment:
Splitting this into multiple lines will help improve readability (currently extending out of view).